File: 1/Scala/CribaEratostenes.scala
object Criba {
    def eratosEstructurado(n:Int):Int = {
       while (x > 1){
            var esprimo = True;
            for (i <- 2 to x){
                if (x % i == 0){
                    esprimo = False;
                    i <- x + 1;
                }
                if (esprimo){
                    print(x);
                    i <- x + 1;
                }
            }
            x <- x - 1;
        }
    }
    
    def eratosRecursivo(n:Int):Int = {
        esprimo = True
        if(x > 1){
            for (i <- 2 to x){
                if x % i == 0 {
                    eratosReursivo(x - 1);
                    esprimo = False;
                    i <- x + 1;
                }
            }
            if (esprimo){
                eratosReursivo(x - 1);
                println(x);
            }
        }
        else {
            return;
        }
    }
    def main(args: Array[String]):Unit = {
        println("Criba Estructurada: ");
        val temp = scala.io.StdIn.readInt();
        eratosEstructurado(temp);
        println("Criba Recursiva: ");
        temp = scala.io.StdIn.readInt();
        eratosRecursivo(temp);
    }   
}

File: 1/Haskell/CribaEratostenes.hs
eratos :: Integer -> Integer
eratos 0 = return()
eratos 1 = return()
eratos n = 
    if n > 1 then
        print(primo 2 n)

primo i n =
    if i <= n then
        if (n % i == 0) then
            eratos(n - 1)
            return ()
        return n
        primo (i + 1) n

main = do
    putStrLn "Criba Recursiva"
    input <- getLine
    let temp = (read input :: Int)
    eratos temp